docs/contributing/ux-guidelines.md

```diff
@@ -3,8 +3,119 @@ id: ux-guidelines
+
+### **MUST** output a list of items to a table by default
+
+When displaying a list of items, it **MUST** be output into a table format by default. The CLI uses a standard table format and provides libraries to do so in consistency with the other tables.
+
+![List of items in table format](../../website/static/img/ux-assets/list-table-format.png)
+
+An `--output` flag **MUST** be provided to enable the user to output the list in either JSON or YAML format.
+
+![List of items in JSON format](../../website/static/img/ux-assets/list-table-format-json.png)
+
+### **SHOULD** display output on success
+
+When a command has finished executing successfully, brief text should be output to update the user.
+
+![Success output message](../../website/static/img/ux-assets/output-success-message.png)
+
+It is **RECOMMENDED** to prefix a success message with a green check mark "✔️" for Unicode compatible shells.
+
+### **SHOULD** display an error message on command fail
+
+When an error has occured during command execution a clear and concise error message **SHOULD** be printed before exiting the program.
+
+![Error output message](../../website/static/img/ux-assets/output-failure-message.png)
+
+### **MAY** use color in human-readable output where it enhances readability
+
+The use of color **MAY** be used if you feel that its additional will improve readability of the output. Avoid overuse of color and limit the range of colors used.
+
+The [`color`](https://pkg.go.dev/github.com/redhat-developer/app-services-cli@v0.37.1-beta3/pkg/core/ioutil/color) package provides a number of APIs which provided common colors according to how they should be used.
+
+### **SHOULD** allow the user to disable color output
+
+User's may need or want color output disabled. When the `NO_COLOR` enviroment variable is set then the additional of ANSI color to the output should be prevented.
+
+### **SHOULD NOT** display any animations in `stdout`
+
+Animations should only be displayed in `stderr`, as `stdout` needs to be kept for machine-readable output.
+
+### **MAY** use symbols and emoji where it makes things clearer
+
+Symbols and emojis can sometimes tell more than words can and they can also be useful to draw someone's attention to something.
+By default all error messages are prefixed with a ❌ to highlight that something went wrong. When a symbol is used more than once it becomes a pattern and should always mean the same thing.
+
+![Error icon pattern](../../website/static/img/ux-assets/error-icon.png)
+
+Be careful and don't overdo it - too many symbols can dilute their significance and can make the application look like a toy.
+
+### **MUST** use color in all JSON and YAML output
+
+All JSON and YAML output **MUST** be syntax highlighted by default.
+
+**JSON:**
+
+![JSON Output example](../../website/static/img/ux-assets/json-output.png)
```

Comment:
Image references are not canonical. In docusaurus they say to use basepath.

so urls will be `/assets/img/ux-assets/json-output.png`

Reply:
As the docs are in the root of the repo outside of the website directory, doing so would break the raw files, if someone would like to view it that way. For example if you open the file directly: https://github.com/craicoverflow/app-services-cli/blob/ux-guidelines/docs/contributing/ux-guidelines.md you can still see the images rendered. It would also be the case when viewing the files while checked out locally in your IDE.

Reply:
Good point. if that works in production then we good